textile/chat-features.textile

```diff
@@ -9,5 +9,148 @@ jump_to:
+* @(CHA-RS2)@ A room must expose its current status.
+** @(CHA-RS2a)@ @[Testable]@ A room must expose its current status, a single value from the list provided above.
+** @(CHA-RS2b)@ A room must expose the latest error, if any, associated with its current status.
+*** @(CHA-RS2b1)@ @[Testable]@ If an error is associated with the current status, then this must be exposed.
+*** @(CHA-RS2b2)@ @[Testable]@ If there is no error is associated with the current status, then the room status should not expose any errors.
+* @(CHA-RS3)@ @[Testable]@ A newly created room must have a current status of @INITIALIZED@.
+* @(CHA-RS4)@ A room must allow changes to room status to be observed by clients.
+** @(CHA-RS4a)@ @[Testable]@ Room status update events must contain the newly entered room status.
+** @(CHA-RS4b)@ @[Testable]@ Room status update events must contain the previous room status.
+** @(CHA-RS4c)@ @[Testable]@ Room status update events must not contain an error, where that error pertains to the current status.
+** @(CHA-RS4d)@ @[Testable]@ Room status update events must contain an error, where that error pertains to the current status.
+** @(CHA-RS4e)@ @[Testable]@ Clients must be able to register a listener for room status updates and receive such events.
+** @(CHA-RS4f)@ @[Testable]@ Clients must be able to unregster a listener for room status updates and from that point onwards, cease to receive such events on that listener only.
+** @(CHA-RS4g)@ @[Testable]@ Clients must be able to unregster all listeners for room status updates and from that point onwards, cease to receive such events on all listeners.
+
+h3(#rooms-lifecycle). Room Lifecycle
+
+Rooms are considered the atomic unit of chat and comprise of potentially multiple underlying realtime channels. The status of the room is, broadly speaking, the combination of the respective underlying channel states (the rooms @contributors@). We present one unified status, rather than statuses for multiple individual features, as this is easier for developers to reason about.
+
+Chat features are spread across multiple Realtime Channels. For the purpose of this section, a @contributor@ is a feature in Chat that is considered as part of the overall room status, as well as its lifecycle. A @contributor@ might share its realtime channel with another (e.g. messages and occupancy), or it might be entirely standalone (room reactions). In that sense, implementations of Room Lifecycle @MUST@ make no assumptions over which realtime channels are in use or being shared, and treat every @contributor@ as being entirely standalone.
+
+There are four room lifecycle operations: @ATTACH@, @DETACH@, @RELEASE@ and @RETRY@. @RETRY@ is an internal-only operation. In order to ensure the integrity of the chat room status, these operations are carried out atomically.
+
+h4(#rooms-lifecycle-operations). Room Lifecycle Operations
+
+* @(CHA-RL1)@ A room must be explicitly @ATTACHED@ to.
```

Comment:
```suggestion
* @(CHA-RL1)@ A room must be explicitly attached via the @ATTACH@ operation.
```

(using `ATTACHED` here is a bit confusing because it's also a state)

Reply:
Applies to CHA-RL2 and CHA-RL3 too

Reply:
Done!